programs/cabal-gild.nix

```diff
@@ -0,0 +1,43 @@
+{
+  mkFormatterModule,
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.cabal-gild;
+in
+{
+  meta.maintainers = [ ];
+
+  imports = [
+    (mkFormatterModule {
+      name = "cabal-gild";
+      package = [
+        "haskellPackages"
+        "cabal-gild"
+      ];
+      includes = [
+        "*.cabal"
+        "*cabal.project"
+        "*cabal.project.local"
+      ];
+    })
+  ];
+
+  config = lib.mkIf cfg.enable {
+    settings.formatter.cabal-gild = {
+      # cabal-gild doesn't support multiple file targets, see upstream issue 35
+      command = pkgs.writeShellApplication {
+        name = "cabal-gild-wrapper";
+        text = ''
+          for file in "$@"; do
+            ${lib.getExe cfg.package} --io="$file"
```

Comment:
Are you sure we don't need `--input="$file"` here? From https://hackage.haskell.org/package/cabal-gild:

> `--input=FILE`: Uses `FILE` as the input. If this is `-` (which is the default), then the input will be read from `STDIN`.

Reply:
Yes, I am sure and it works well in my projects.

I set `--io`and here is the doc:

--io=FILE: Shortcut for setting both --input=FILE and --output=FILE at the same time. This is useful for formatting a file in-place.